docs/docs/functional-agents.md

```diff
@@ -0,0 +1,282 @@
+
+# Functional agents
+
+Functional agents are lightweight AI agents that operate without building complex strategy graphs.
+Instead, the agent logic is implemented as a lambda function that handles user input, interacts with an LLM,
+optionally calls tools, and produces a final output. It can perform a single LLM call,
+manage multiple LLM calls in a sequence, or repeat in a loop when needed based on user input and LLM and tool outputs.
```

Comment:
Let's add about the reasoning here:
It's recommended to start with the out-of-the-box simple agent -------> then you can easily prototype the logic using functional agents by just writing plain Kotlin code because it's east and feels natural -- but still with the access to most of the Koog's features, including history compression and automatic state management ------> and finally, if you need Persistence and checkpointing for fault-tolerance, and better OpenTelemetry tracing with nested graph events -- you can refactor your agents to graph strategies and get the most reliability that Koog can offer!